Lib/xml/etree/ElementPath.py

```diff
@@ -253,13 +254,17 @@ def select(context, result):
                 if elem.get(key) is not None:
                     yield elem
         return select
-    if signature == "@-='":
-        # [@attribute='value']
+    if signature == "@-='" or signature == "@-!='":
+        # [@attribute='value'] or [@attribute!='value']
         key = predicate[1]
         value = predicate[-1]
+        negate_predicate = '!' in signature
         def select(context, result):
```

Comment:
Please use separate functions for performance reasons. They are deeply nested and a single instruction less can make a visible difference on large enough data. This implementation probably adds 4 byte code instructions to the inner loop.

Reply:
Gotcha, added a separate function for the negated conditions.